libraries/bezier-rs/src/bezier/lookup.rs

```diff
@@ -45,28 +22,54 @@ impl Bezier {
 				high = mid;
 			}
 		}
-		self.unrestricted_parametric_evaluate(mid)
+
+		mid
 	}
 
-	/// Calculate the point on the curve based on the `t`-value provided.
-	/// Expects `t` to be within the inclusive range `[0, 1]`.
-	pub fn evaluate(&self, t: ComputeType) -> DVec2 {
-		match t {
+	/// Convert a [ComputeType] to a parametric `t`-value.
+	pub(crate) fn compute_type_to_parametric(&self, c: ComputeType) -> f64 {
+		match c {
 			ComputeType::Parametric(t) => {
 				assert!((0.0..=1.).contains(&t));
-				self.unrestricted_parametric_evaluate(t)
+				t
 			}
-			ComputeType::Euclidean(t) => {
-				assert!((0.0..=1.).contains(&t));
-				self.unrestricted_euclidean_evaluate(t, 0.0001)
+			ComputeType::Euclidean(distance) => {
+				assert!((0.0..=1.).contains(&distance));
+				self.euclidean_to_parametric(distance, DEFAULT_EUCLIDEAN_ERROR_BOUND)
 			}
-			ComputeType::EuclideanWithinError { t, epsilon } => {
-				assert!((0.0..=1.).contains(&t));
-				self.unrestricted_euclidean_evaluate(t, epsilon)
+			ComputeType::EuclideanWithinError { distance, epsilon } => {
+				assert!((0.0..=1.).contains(&distance));
+				self.euclidean_to_parametric(distance, epsilon)
 			}
 		}
 	}
 
+	/// Calculate the point on the curve based on the `t`-value provided.
+	pub(crate) fn unrestricted_parametric_evaluate(&self, t: f64) -> DVec2 {
```

Comment:
This name that appears in many functions, "unrestricted", doesn't have a clear meaning to me. Can you suggest some alternatives in this comment's discussion so we can settle on a better name throughout the code?

Reply:
We use `unrestricted` for functions which may perform some computation or return some value in which the `t`-values used are not bounded by [0., 1.]. We could convert this naming convention to `unbounded` if that is preferred,